pkg/controller/direct/externalresource/resource.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package externalresource
+
+import (
+	"reflect"
+)
+
+type ExternalResourceReference struct {
+	hasSelfLink bool
+	externalRef string
+}
+
+type GCPResource interface {
+	GetName() string
+}
+
+// New builds the External Reference for CloudBuildWorkerPool when the object already exists on the GCP server.
+// Ideally, it expects to use the GCP object's selfLink value as the external reference. However, if
+// the resource does not have `selfLink` (like CloudBuildWorkerPool), it builds the URL from the GCP object.
+func New(host string, gcpObj GCPResource) *ExternalResourceReference {
+	baseExternalRef := &ExternalResourceReference{
+		hasSelfLink: false,
+		externalRef: host + gcpObj.GetName(),
```

Comment:
Any better name than "host"? This externalRef is to make the value the same as the `seflLink`.

Reply:
Thanks! I like serviceBaseURL